tasks/managed/push-rpm-data-to-pyxis/push-rpm-data-to-pyxis.yaml

```diff
@@ -180,33 +180,112 @@ spec:
           IMAGEURL=$(jq -r '.containerImage' <<< "${COMPONENT}")
           NUM_PYXIS_IMAGES=$(jq '.pyxisImages | length' <<< "${COMPONENT}")
-          # cosign has very limited support for selecting the right auth entry,
-          # so create a custom auth file with just one entry
-          select-oci-auth "$IMAGEURL" > "$DOCKER_CONFIG"/config.json
+          
           for (( j=0; j < NUM_PYXIS_IMAGES; j++ )); do
             PYXIS_IMAGE=$(jq -c --argjson j "$j" '.pyxisImages[$j]' <<< "${COMPONENT}")
-            FILE="$(jq -r '.imageId' <<< "$PYXIS_IMAGE").json"
-            DIGEST="$(jq -r '.digest' <<< "$PYXIS_IMAGE")"
-            ARCH_DIGEST="$(jq -r '.arch_digest' <<< "$PYXIS_IMAGE")"
-            # You can't pass --platform to a single arch image or cosign errors.
-            # If digest equals arch_digest, then it's a single arch image
-            if [ "$DIGEST" = "$ARCH_DIGEST" ] ; then
-              echo "Fetching sbom for single arch image: $IMAGEURL to: $FILE"
-              run_cosign download sbom --output-file "${FILE}" "${IMAGEURL}"
-            else
-              OS=$(jq -r '.os' <<< "$PYXIS_IMAGE")
-              ARCH=$(jq -r '.arch' <<< "$PYXIS_IMAGE")
-              PLATFORM="${OS}/${ARCH}"
-              echo "Fetching sbom for image: $IMAGEURL with platform: $PLATFORM to: $FILE"
-              run_cosign download sbom --output-file "${FILE}" --platform "${PLATFORM}" "${IMAGEURL}"
-            fi
+            IMAGEID=$(jq -r '.imageId' <<< "$PYXIS_IMAGE")
+            
+            # Store task information for parallel execution
+            download_tasks+=("$PYXIS_IMAGE")
+            task_images+=("$IMAGEID")
+            task_urls+=("$IMAGEURL")
           done
         done
 
+        # Process downloads in parallel batches
+        N=$(params.concurrentLimit)  # The maximum number of images to be processed at once
+        declare -a jobs=()
+        total=${#download_tasks[@]}
+        count=0
+        success=true
+        echo "Starting SBOM download for $total images in total. " \
+          "Up to $N images will be downloaded at once..."
+
+        for (( idx=0; idx < total; idx++ )); do
+          PYXIS_IMAGE="${download_tasks[idx]}"
+          IMAGEID="${task_images[idx]}"
+          IMAGEURL="${task_urls[idx]}"
+          
+          echo "Starting download for IMAGE: $IMAGEID from URL: $IMAGEURL"
+          download_sbom_for_image "$PYXIS_IMAGE" "$IMAGEURL" > "${IMAGEID}.download.out" 2>&1 &
+          
+          jobs+=($!)  # Save the background process ID
+          ((++count))
+
+          if [ $((count%N)) -eq 0 ] || [ $((count)) -eq "$total" ]; then
+            echo "Waiting for the current batch of background processes to finish"
+            for job_id in "${!jobs[@]}"; do
+              if ! wait "${jobs[job_id]}"; then
+                echo "Error: download of sbom failed for one of the images"
+                success=false
+              fi
+            done
```

Comment:
So this will always wait for the whole batch to finish before starting a new one, right? What we have elsewhere is that we the number of running jobs full - as soon as one finishes, we add another one. I guess in practice that won't make too much of a difference, but you can have one large sbom that would be blocking the whole batch. So I wonder if it would make sense to use that same approach here. Here's an example: https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/managed/create-pyxis-image/create-pyxis-image.yaml#L395